pkg/skaffold/runner/v1/dev.go

```diff
@@ -226,6 +230,32 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 			continue
 		}
 
+		var absMatches []string
```

Comment:
The bulk of this section should be moved further down and called from the deps function called by `r.monitor.Register(...)` registration (the one that calls `TransitiveArtifactDependencies()`).  These absolute rules could have globs and the matching file sets could change over the course of the dev session.

It would make sense to move this expansion into a helper function.

It does make sense to iterate through all of the sync rules and validate that all globs are valid, not just the absolute file path globs.  That should be done in the [sync-rule validation](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/schema/validation/validation.go#L442-L443); the easiest way seems to be to call `doublestar.Match(patt, "/").

Reply:
I definitely need help on that. You suggested many thinks and I don't know which to do.